src/nspawn/nspawn.c

```diff
@@ -3206,6 +3206,17 @@ static int outer_child(
         return 0;
 }
 
+static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        pid_t pid;
+
+        pid = PTR_TO_UINT32(userdata);
+        if (pid == si->ssi_pid) {
```

Comment:
@aulanov , OK, this PR fixes https://github.com/systemd/systemd/pull/4855/files#r91656327
But we don't reap zombies:
```
root     28849  0.0  0.2 118836  5812 pts/1    S+   05:12   0:00      \_ sudo ../../libtool --mode=execute ../../systemd-nspawn -D ./CONT/ --share-system
root     28850  0.4  0.1  52480  2668 pts/1    S+   05:12   0:00          \_ /home/vagrant/systemd/.libs/lt-systemd-nspawn -D ./CONT/ --share-system
root     28881  0.0  0.0   1272     4 ?        Ss   05:12   0:00              \_ -sh
root     28882  0.0  0.0      0     0 ?        Z    05:12   0:00              |   \_ [sh] <defunct>
root     28883  0.0  0.0      0     0 ?        Z    05:12   0:00              \_ [sh] <defunct>
```
Not sure this is OK